pkg/service/nodes.go

```diff
@@ -128,32 +128,39 @@ func labelNode(kubectlPath, nodeName, labelKey, labelValue string) error {
 	return nil
 }
 
-func annotateNode(kubectlPath, nodeName, key, value string) error {
-	label := fmt.Sprintf("%v=%v", key, value)
-	labelArgs := []string{"annotate", "--overwrite", "node", nodeName, label}
-	_, err := runCommand(kubectlPath, labelArgs)
+func annotateNode(kubectlPath string, nodeName string, annotations map[string]string) error {
+	annotateArgs := []string{"annotate", "--overwrite", "node", nodeName}
+	for k, v := range annotations {
+		annotateArgs = append(annotateArgs, fmt.Sprintf("%v=%v", k, v))
+	}
+	_, err := runCommand(kubectlPath, annotateArgs)
 	if err != nil {
 		log.Errorf("failed to annotate node %v", nodeName)
 		return err
 	}
 	return nil
 }
 
-func getNodesByAnnotationKey(kubeClient kubernetes.Interface, key string) (map[string]string, error) {
-	results := make(map[string]string, 0)
+func getNodesByAnnotationKeys(kubeClient kubernetes.Interface, keys ...string) (map[string]map[string]string, error) {
+	results := make(map[string]map[string]string, 0)
 
 	nodes, err := kubeClient.CoreV1().Nodes().List(metav1.ListOptions{})
 	if err != nil {
 		return results, err
 	}
 
+NODES:
 	for _, node := range nodes.Items {
 		annotations := node.GetAnnotations()
-		for k, v := range annotations {
-			if k == key {
-				results[node.Name] = v
+		resultValues := make(map[string]string)
+		for _, k := range keys {
+			if v, ok := annotations[k]; ok {
+				resultValues[k] = v
+			} else {
+				continue NODES
```

Comment:
Seems this will skip the node if the one of the annotations are missing, this might cause a problem if we upgrade the pod while a node is terminating e.g. a node is in-progress, then the lifecycle-manager image is replaced to this version, then it tries to resume the instance, which might not have the queue annotation and skip. It should be OK to go over all annotations (we are only checking for existence of two) and avoid the continue statement, this will also allow us to refactor this a bit to also get rid of the go-to statement, which makes it a bit less readable.